.github/workflows/ci.yaml

```
@@ -71,20 +74,24 @@ jobs:
   # optimizer falls into the run-CI bucket, never the skip bucket.
   #
-  # NOTE: paths-filter needs the repo checked out for push/workflow_dispatch git
-  # change detection (this workflow is not pull_request-triggered).
+  # NOTE: paths-filter uses the Pull Requests API for pull_request runs and git
+  # change detection for push/workflow_dispatch, so the checkout is needed for
+  # the latter two events.
   changes:
     name: changes
     needs: optimize_ci
     if: ${{ always() && needs.optimize_ci.outputs.skip != 'true' }}
     runs-on: blacksmith-2vcpu-ubuntu-2404
```

actionlint (1.7.12)

[error] label "blacksmith-2vcpu-ubuntu-2404" is unknown. available labels are "windows-latest", "windows-latest-8-cores", "windows-2025", "windows-2025-vs2026", "windows-2022", "windows-11-arm", "ubuntu-slim", "ubuntu-latest", "ubuntu-latest-4-cores", "ubuntu-latest-8-cores", "ubuntu-latest-16-cores", "ubuntu-24.04", "ubuntu-24.04-arm", "ubuntu-22.04", "ubuntu-22.04-arm", "macos-latest", "macos-latest-xlarge", "macos-latest-large", "macos-26-intel", "macos-26-xlarge", "macos-26-large", "macos-26", "macos-15-intel", "macos-15-xlarge", "macos-15-large", "macos-15", "macos-14-xlarge", "macos-14-large", "macos-14", "self-hosted", "x64", "arm", "arm64", "linux", "macos", "windows". if it is a custom label for self-hosted runner, set list of labels in actionlint.yaml config file

(runner-label)